src/handlers/push_message.rs

```diff
@@ -54,13 +55,92 @@ pub async fn handler(
     headers: HeaderMap,
     RequireValidSignature(Json(body)): RequireValidSignature<Json<PushMessageBody>>,
 ) -> Result<Response> {
-    let request_id = get_req_id(&headers);
+    let (res, analytics_option) = handler_internal(
+        Path((tenant_id, id)),
+        StateExtractor(state),
+        headers,
+        RequireValidSignature(Json(body)),
+    )
+    .await;
 
-    increment_counter!(state.metrics, received_notifications);
+    let (status, response) = match res {
+        Ok(res) => (res.status_code.clone().as_u16(), res),
+        Err(error) => {
+            let res = error.into_response();
+            let status_code = res.status().clone().as_u16();
+            (status_code, res)
+        }
+    };
+
+    #[cfg(feature = "analytics")]
+    if let Some(mut message_info) = analytics_option {
+        message_info.status = status;
+        message_info.success = status >= 200 && status < 300;
+        message_info.response_message = Some(response.body().to_string());
```

Comment:
Is the presence of 'response_message' the only differentiator between a request and a response?
In that case, can we have a field that tells explicitly which is which to avoid potential downstream issues?

Reply:
response message was so we could provide context on errors

Reply:
ok, I think we still need to be able to differentiate between requests and responses